frontend/src/components/CreateGroup/CreateGroup.jsx

```diff
@@ -0,0 +1,151 @@
+import { createGroup } from "../../utils/groups";
+import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { useState } from "react";
+import "./CreateGroup.css"
+
+const CreateGroup = () => {
+    document.querySelector('title').innerText = 'Start a New Group';
+    const navigate = useNavigate();
+    const dispatch = useDispatch();
+    const [location, setLocation] = useState('');
+    const [name, setName] = useState('');
+    const [about, setAbout] = useState('');
+    const [type, setType] = useState('');
+    const [isPrivate, setIsPrivate] = useState('');
+    const [url, setUrl] = useState('');
+    const [errors, setErrors] = useState({});
+
+    const handleSubmit = async e => {
+        e.preventDefault();
+        setErrors({});
+        const locationArray = location.split(', ');
```

Comment:
maybe add some error handling to check if the user followed the correct format for the location?

Reply:
looks good! you can go even further by checking the length of state element to make sure it is 2 chars long, or you could even convert it to a dropdown menu with all the valid state abbreviations. Although what you have now is probably good enough.